src/hotspot/share/cds/lambdaFormInvokers.cpp

```diff
@@ -46,27 +47,56 @@
 #include "runtime/handles.inline.hpp"
 #include "runtime/javaCalls.hpp"
 
-GrowableArray<char*>* LambdaFormInvokers::_lambdaform_lines = NULL;
+GrowableArrayCHeap<char*, mtClassShared>* LambdaFormInvokers::_lambdaform_lines = nullptr;
+Array<Array<char>*>* LambdaFormInvokers::_static_archive_invokers = nullptr;
+#define NUM_FILTER 4
+static const char* filter[NUM_FILTER] = {"java.lang.invoke.Invokers$Holder",
+                                         "java.lang.invoke.DirectMethodHandle$Holder",
+                                         "java.lang.invoke.DelegatingMethodHandle$Holder",
+                                         "java.lang.invoke.LambdaForm$Holder"};
 
 void LambdaFormInvokers::append(char* line) {
   if (_lambdaform_lines == NULL) {
-    _lambdaform_lines = new GrowableArray<char*>(100);
+    _lambdaform_lines = new GrowableArrayCHeap<char*, mtClassShared>(150);
   }
   _lambdaform_lines->append(line);
 }
 
+void LambdaFormInvokers::append_filtered(char* line) {
+  for (int k = 0; k < NUM_FILTER; k++) {
+    if (strstr(line, filter[k]) != nullptr) {
+        append(line);
+        break;
+    }
+  }
+}
+#undef NUM_FILTER
+
 void LambdaFormInvokers::regenerate_holder_classes(TRAPS) {
-  assert(_lambdaform_lines != NULL, "Bad List");
+  if (_lambdaform_lines == nullptr || _lambdaform_lines->length() == 0) {
+    log_info(cds)("Nothing to regenerate for holder classes");
+    return;
+  }
   ResourceMark rm(THREAD);
 
   Symbol* cds_name  = vmSymbols::jdk_internal_misc_CDS();
   Klass*  cds_klass = SystemDictionary::resolve_or_null(cds_name, THREAD);
   guarantee(cds_klass != NULL, "jdk/internal/misc/CDS must exist!");
 
+  log_info(cds)("Total lambdaform lines %d", _lambdaform_lines->length());
+  HandleMark hm(THREAD);
   int len = _lambdaform_lines->length();
   objArrayHandle list_lines = oopFactory::new_objArray_handle(vmClasses::String_klass(), len, CHECK);
+  if (list_lines == nullptr) {
```

Comment:
There's no need to check for null. If allocation fails, the CHECK in the previous line will force this function to return.
(same for the check on line 96).